psl-core/src/main/java/org/linqs/psl/model/rule/Rule.java

```diff
@@ -38,4 +38,14 @@
 	public int groundAll(AtomManager atomManager, GroundRuleStore groundRuleStore);
 
 	public boolean isWeighted();
+
+	/**
+	 * @return a name for the rule
+	 */
+	public String getName();
+
+	/**
+	 * @param name a name for the rule
+	 */
+	public void setName(String name);
 }
```

Comment:
Should we also add the rule's name to its toString()?
Actually, don't do that.
As it currently is, a rule's toString() produces a syntactically valid string rule (so it can be directly pasted into a string rule or CLI).
But maybe some sort of pretty to string or like a toString(bool pretty).